manifests/backports.pp

```diff
@@ -0,0 +1,41 @@
+class apt::backports (
+  $location = undef,
+  $release  = undef,
+  $repos    = undef,
+  $key      = undef,
+  $pin      = 200,
+){
+
+  if ($::apt::xfacts['lsbdistid'] == 'debian' or
+      $::apt::xfacts['lsbdistid'] == 'ubuntu') {
+    $_location = $::apt::backports['location']
+    $_release  = "${::apt::xfacts['lsbdistcodename']}-backports"
+    $_repos    = $::apt::backports['repos']
+    $_key      = $::apt::backports['key']
```

Comment:
Probably. I'm just not sure how to do it in a clean way that doesn't get in your way if you're on Debian/Ubuntu but does by default nag at you if you're not.

I considered binding it as a parameter but then you have to go like this:
```puppet
if (($::apt::xfacts['lsbdistid'] != 'debian' or
   $::apt::xfacts['lsbdistid'] != 'ubuntu') and
   $location = $::apt::backports['location']) {
    fail('If you're not on Debian/Ubuntu you need to explicitly pass location')
}
```

It gets very scary.